beacon_node/network/src/sync/block_lookups/single_block_lookup.rs

```diff
@@ -343,336 +197,290 @@ impl<const MAX_ATTEMPTS: u8, T: BeaconChainTypes> SingleBlockLookup<MAX_ATTEMPTS
+    /// necessary and clear the blob cache.
+    pub fn handle_consistency_failure(&mut self, cx: &SyncNetworkContext<T>) {
+        self.penalize_lazy_blob_peer(cx);
+        if let Some(cached_child) = self.cached_child_components.as_mut() {
+            cached_child.clear_blobs();
         }
+        self.blob_request_state.state.register_failure_downloading()
     }
 
-    fn get_peer(&self, response_type: ResponseType) -> Option<PeerShouldHave> {
-        match response_type {
-            ResponseType::Block => self
-                .block_request_state
-                .state
-                .available_peers
-                .iter()
-                .choose(&mut rand::thread_rng())
-                .copied()
-                .map(PeerShouldHave::BlockAndBlobs)
-                .or(self
-                    .block_request_state
-                    .state
-                    .potential_peers
-                    .iter()
-                    .choose(&mut rand::thread_rng())
-                    .copied()
-                    .map(PeerShouldHave::Neither)),
-            ResponseType::Blob => self
-                .blob_request_state
-                .state
-                .available_peers
-                .iter()
-                .choose(&mut rand::thread_rng())
-                .copied()
-                .map(PeerShouldHave::BlockAndBlobs)
-                .or(self
-                    .blob_request_state
-                    .state
-                    .potential_peers
-                    .iter()
-                    .choose(&mut rand::thread_rng())
-                    .copied()
-                    .map(PeerShouldHave::Neither)),
+    /// This failure occurs after processing, so register a failure processing, penalize the peer if
+    /// necessary and clear the blob cache.
+    pub fn handle_availability_check_failure(&mut self, cx: &SyncNetworkContext<T>) {
+        self.penalize_lazy_blob_peer(cx);
+        if let Some(cached_child) = self.cached_child_components.as_mut() {
+            cached_child.clear_blobs();
         }
+        self.blob_request_state.state.register_failure_processing()
     }
+}
 
-    fn add_used_peer(&mut self, peer_id: PeerShouldHave, response_type: ResponseType) {
-        match response_type {
-            ResponseType::Block => {
-                self.block_request_state
-                    .state
-                    .used_peers
-                    .insert(peer_id.to_peer_id());
-                self.block_request_state.state.state = State::Downloading { peer_id };
-            }
-            ResponseType::Blob => {
-                self.blob_request_state
-                    .state
-                    .used_peers
-                    .insert(peer_id.to_peer_id());
-                self.blob_request_state.state.state = State::Downloading { peer_id };
-            }
+/// The state of the blob request component of a `SingleBlockLookup`.
+pub struct BlobRequestState<L: Lookup, T: EthSpec> {
```

Comment:
Unrelated note 😅 , I'm a bit tempted to consider replacing`T: EthSpec` with `E: EthSpec` globally, I always get it mixed up with the other `T` we commonly see in our code base:`T: BeaconChainTypes`

Reply:
My usage of the two switches on a dime and I confuse myself